MeesGame/Gameplay/GameObjects/TileField.cs

```diff
@@ -135,15 +135,41 @@ public void UpdateGraphics()
 
         public void RevealArea(Point a)
         {
+            revealAroundTile(a);
+
+            revealPath(a, new int[2] { -1, 0 });
+            revealPath(a, new int[2] { 1, 0 });
+            revealPath(a, new int[2] { 0, 1 });
+            revealPath(a, new int[2] { 0, -1 });
+        }
+
+        public void revealAroundTile(Point a)
+        {
             for (int i = a.X - 1; i <= a.X + 1; i++)
             {
                 for (int j = a.Y - 1; j <= a.Y + 1; j++)
                 {
-                    if (i >= 0 && i < this.Columns && j >= 0 && j < this.Rows) ((Tile)this.grid[i, j]).Revealed = true;
+                    if (i >= 0 && i < this.Columns && j >= 0 && j < this.Rows)
+                    {
+                        //using bitwise or-operator
+                        if (i <= a.X && j >= a.Y) { GetTile(i, j).Revealed = GetTile(i, j).Revealed | 1; }
+                        if (i <= a.X && j <= a.Y) { GetTile(i, j).Revealed = GetTile(i, j).Revealed | 2; }
```

Comment:
Just as with + and *, the operator |= exists. That should make it even easier. Read to other comments for an even better solution.